backend/s3_backend.go

```diff
@@ -161,6 +165,15 @@ func (s *S3Backend) Open(db, version, file string) (io.ReadCloser, error) {
 	}
 	resp, err := s.svc.GetObject(params)
 
+	// If the download failed, retry maxRetries number of times with an
```

Comment:
Oh yeah that makes sense - wouldn't we also want to retry on NoSuchKey? I don't exactly remember the particular problem that motivated this change but I thought that it had to do with the fact that S3 had sometimes not updated the set of keys available after writes occurred.